presto-router/src/main/java/com/facebook/presto/router/spec/GroupSpec.java

```diff
@@ -18,22 +18,32 @@
 import com.google.common.collect.ImmutableList;
 
 import java.net.URI;
+import java.util.ArrayList;
 import java.util.List;
 
+import static java.util.Collections.nCopies;
 import static java.util.Objects.requireNonNull;
 
 public class GroupSpec
 {
     private final String name;
     private final List<URI> members;
+    private final List<Integer> weights;
 
     @JsonCreator
     public GroupSpec(
             @JsonProperty("name") String name,
-            @JsonProperty("members") List<URI> members)
+            @JsonProperty("members") List<URI> members,
+            @JsonProperty("weights") List<Integer> weights)
     {
         this.name = requireNonNull(name, "name is null");
         this.members = ImmutableList.copyOf(requireNonNull(members, "members is null"));
+        if (weights != null) {
```

Comment:
is it possible to use optional<list<int>> for weight?

Reply:
Same as above. I updated the type to `Optional<List<Integer>>` as you suggested.